StyleCop.Analyzers/StyleCop.Analyzers/Helpers/IndentationHelper.cs

```diff
@@ -92,19 +91,23 @@ public static SyntaxTrivia GenerateWhitespaceTrivia(IndentationOptions indentati
 
         private static int GetIndentationSteps(IndentationOptions indentationOptions, SyntaxTriviaList leadingTrivia)
         {
-            var indentationString = string.Empty;
-            for (var i = leadingTrivia.Count - 1; (i >= 0) && leadingTrivia[i].IsKind(SyntaxKind.WhitespaceTrivia); i--)
+            SyntaxTriviaList.Reversed reversed = leadingTrivia.Reverse();
+            int indentationCount = 0;
+
+            foreach (SyntaxTrivia trivia in reversed)
             {
-                indentationString = string.Concat(leadingTrivia[i].ToFullString(), indentationString);
+                if (!trivia.IsKind(SyntaxKind.WhitespaceTrivia))
+                {
+                    break;
+                }
+
+                foreach (char c in trivia.ToFullString())
+                {
+                    indentationCount += c == '\t' ? indentationOptions.TabSize : 1;
```

Comment:
:bulb: The behavior of this is wrong for text like `" \t \t \t \t"` (written as a string literal just so you can see the special characters clearly). Seems like if we're already rewriting the algorithm we could find the issues and fix them too. What do you think?